test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -184,6 +184,53 @@ def test_tf_serving_version_cpu(tensorflow_inference):
 
     stop_and_remove_container(container_name, ctx)
 
+# Added by hballuru to confirm tf-serving-api version is matching tf serving version
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.model("N/A")
+@pytest.mark.canary("Run non-gpu tf serving version test regularly on production images")
+def test_tf_serving_api_version_cpu(tensorflow_inference):
+    """
+    For non-huggingface non-GPU TF inference images, check that the tag version matches the version of TF serving api
+    in the container.
+
+    Huggingface includes MMS and core TF, hence the versioning scheme is based off of the underlying tensorflow
+    framework version, rather than the TF serving version.
+
+    GPU inference images will be tested along side `test_framework_and_cuda_version_gpu` in order to be judicious
+    about GPU resources. This test can run directly on the host, and thus does not require additional resources
+    to be spun up.
+
+    @param tensorflow_inference: ECR image URI
+    """
+    # Set local variable to clarify contents of fixture
+    image = tensorflow_inference
+
+    if "gpu" in image:
+        pytest.skip(
+            "GPU images will have their framework version tested in test_framework_and_cuda_version_gpu")
+    if "neuron" in image:
+        pytest.skip(
+            "Neuron images will have their framework version tested in test_framework_and_neuron_sdk_version")
+
+    _, tag_framework_version = get_framework_and_version_from_tag(
+        image)
+
+    image_repo_name, _ = get_repository_and_tag_from_image_uri(image)
+
+    if re.fullmatch(r"(pr-|beta-|nightly-)?tensorflow-inference", image_repo_name) and Version(tag_framework_version) == Version("2.6.3"):
+        pytest.skip("Skipping this test for TF 2.6.3 inference as the v2.6.3 version is already on production")
+
+    ctx = Context()
+    container_name = get_container_name("tf-serving-api-version", image)
+    start_container(container_name, image, ctx)
+    output = run_cmd_on_container(
+        container_name, ctx, "pip show tensorflow-serving-api | grep Version", executable="bash"
+    )
+    str_version_from_output = ((str(output.stdout).split(' '))[1]).strip()
+    assert (tag_framework_version == str_version_from_output), \
+        f"Tensorflow serving API version is {str_version_from_output} while the Tensorflow version is {tag_framework_version}. Both don't match!"
+
+    stop_and_remove_container(container_name, ctx)
```

Comment:
Can we use a `try` and `final` strategy to ensure that `stop_and_remove_container` is executed even if the assert statement fails?

Reply:
try catch finally has been implemented to ensure container removal. However bear in mind there are other parts of the same test where the container removal is not enclosed within try block.